insights/tests/test_fava.py

```diff
@@ -808,16 +815,57 @@ def test_yaml_tests(yaml2jsonTest):
     assert yaml2jsonTest[1] == fava.FavaRule.yaml_deserialize(yaml2jsonTest[0])
 
 
+PARSER_NAMES_TEST_TEMPLATE = string.Template("""
+---
+rule:
+  name: "PARSER_NAME_${name}s_TEST_KEY"
+  pydata:
+    kernel: "{{ ${name} }}"
+  when: True
+""")
+
+
+PARSERS_TO_TEST = [
+    Uname,
+    InstalledRpms,
+    FSTab,
+    IfCFG,
+    Ethtool,
+]
+
+
+@pytest.fixture(params=PARSERS_TO_TEST)
+def each_parser_to_test(request):
+    return request.param
+
+
+def test_parser_names_test(each_parser_to_test):
+    # this should not raise exception
+    fava.FavaRule(PARSER_NAMES_TEST_TEMPLATE.substitute({"name": each_parser_to_test.__name__}))
+
+
+def test_parser_names_nosuchparser():
+    # arbitrary names should not work in Fava
+    with pytest.raises(fava.FavaTranslationError):
+        fava.FavaRule(PARSER_NAMES_TEST_TEMPLATE.substitute({"name": "Nosuchparser"}))
+
+
+def test_parser_names_combiner():
+    # combiner names should not work in Fava
+    with pytest.raises(fava.FavaTranslationError):
+        fava.FavaRule(PARSER_NAMES_TEST_TEMPLATE.substitute({"name": hostname.__name__}))
+
+
```

Comment:
I would rather covert the whole file as a separate PR, than just the code below in this PR.  

Would that be acceptable?